### ISSUE TITLE
paths require '/' in order to match

### ISSUE THREAD

**Issue by jkingyens** [codemix/gitignore-parser#1]
Why do lines in my .gitignore file require a leading '/' in order to match?

For example, I usually just add lines like this to my .gitignore:

node_modules

Your module will not match this as an ignore pattern. I need to add a leading '/' in order for this to work:

/node_modules

**Comment by bahamas10** [comment]
@jkingyens @phpnode i'm able to recreate this simply

``` js
var parser = require('./');
var gi;

// simple gitignore with only one line
gi = parser.compile('node_modules');
console.dir(gi.accepts('node_modules'));
// => true

// simple gitignore with multiple newlines
gi = parser.compile('\nnode_modules\n');
console.dir(gi.accepts('node_modules'));
// => true

// simple gitignore with multiple newlines and slashes
gi = parser.compile('\n/node_modules\n');
console.dir(gi.accepts('/node_modules'));
// => true
```

**Comment by sabhiram** [comment]
I took a quick look at the lib code, and here is what I think is going on:

1. Since you have no negative regex patterns - you end up with a default "negative" regex which looks something like this: `/^(())/` which is going to match true for pretty much anything
2. The condition to accepts is `negatives[0].test(input) || !positives[0].test(input);`. Here the positive condition will correctly match `node_modules` and return true (which is negated). But the negative condition always returns true.
3. To fix this, check that the negative match is empty before allowing it to participate in the accepts clause.

**Comment by sabhiram** [comment]
I also think you need to enforce some ordering between the negated and non negated options. They are allowed to vary in specificity.

**Comment by johnhaley81** [comment]
I think this is fixed via #7
